### PR TITLE
Feat/support custom message type

### PR DESCRIPTION
**Pull request by seba-aln**
feat: Add custom message type support for selected APIs

Add custom message type support for the following APIs - publish, signal, share file, subscribe and history

### CODE REVIEW

File: examples/cli/pub.php [hunk 1]
```diff
@@ -9,18 +9,25 @@
 
 use PubNub\PubNub;
 use PubNub\PNConfiguration;
-use PubNub\CryptoModule;
+
+if ($argc < 3) {
```
Comment (marcin-cebo) [inline examples/cli/pub.php:13]: Shouldn't it be 5 instead of 3?

File: examples/cli/hist.php [hunk 2]
```diff
@@ -15,24 +15,30 @@
 $pnConfig = new PNConfiguration();
 $pnConfig->setPublishKey(getenv('PN_KEY_PUBLISH'));
```
Comment (marcin-cebo) [inline examples/cli/hist.php:16]: if PN_KEY_PUBLISH or PN_KEY_SUBSCRIBE is not set, the script might fail without a clear error message ?

File: src/PubNub/Endpoints/MessagePersistance/FetchMessages.php [hunk 3]
```diff
@@ -31,8 +31,9 @@ class FetchMessages extends Endpoint
 
     protected bool $includeMeta = false;
     protected bool $includeUuid = false;
-    protected bool $includeMessageType = false;
+    protected bool $includeMessageType = true;
```
Comment (marcin-cebo) [inline src/PubNub/Endpoints/MessagePersistance/FetchMessages.php:34]: Why you changed default from false to true ?

File: tests/Acceptance/CustomMessageType/history-custom-mssg-type.feature [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Feature: History for VSP
+  As a PubNub user I want to fetch history with message type.
+  Client should be able to opt-out default `includeType`.
+
+  Background:
+    Given the demo keyset with enabled storage
+
+  @contract=fetchHistoryWithPubNubMessageTypes
+  Scenario: Client can fetch history with message types
+    When I fetch message history for 'simple-channel' channel
```
Comment (marcin-cebo) [inline tests/Acceptance/CustomMessageType/history-custom-mssg-type.feature:11]: Check sdk-specification. It has different content e.g. for this line there is:
When I fetch message history with messageType for 'simple-channel' channel

File: examples/cli/pub.php (hunk 1 above)
Comment (seba-aln) [reply examples/cli/pub.php:13]: you are right. missed that

File: examples/cli/hist.php (hunk 2 above)
Comment (seba-aln) [reply examples/cli/hist.php:16]: this is just a code example to be analyzed and not a demo :)

File: tests/Acceptance/CustomMessageType/history-custom-mssg-type.feature (hunk 4 above)
Comment (seba-aln) [reply tests/Acceptance/CustomMessageType/history-custom-mssg-type.feature:11]: this file should not be included in PR... I'll fix it

File: examples/cli/pub.php (hunk 1 above)
Comment (seba-aln) [reply examples/cli/pub.php:13]: my miss. I'll fix it

**Review by marcin-cebo — APPROVED** [review]

**Comment by seba-aln** [comment]
@pubnub-release-bot release

**Comment by seba-aln** [comment]
@pubnub-release-bot release

**Comment by seba-aln** [comment]
@pubnub-release-bot release

**Comment by seba-aln** [comment]
@pubnub-release-bot release

**Review by techwritermat — APPROVED** [review]

**Pull request merged**